lib/miq_automation_engine/engine/miq_ae_method.rb

```diff
@@ -198,32 +180,16 @@ def self.verbose_rc(rc)
     end
 
     def self.run_ruby_method(body, preamble = nil)
-      begin
-        require 'open4'
-        ActiveRecord::Base.connection_pool.release_connection
-
-        rc = nil
-        final_stderr = nil
-        Bundler.with_clean_env do
-          status = Open4.popen4(Gem.ruby) do |pid, stdin, stdout, stderr|
-            stdin.puts(preamble.to_s)
-            stdin.puts(body)
-            stdin.puts(RUBY_METHOD_POSTSCRIPT) unless preamble.blank?
-            stdin.close
-
-            final_stderr = stderr.each_line.map(&:strip)
-            stdout.each_line { |msg| $miq_ae_logger.info  "Method STDOUT: #{msg.strip}" }
-            final_stderr.each { |msg| $miq_ae_logger.error "Method STDERR: #{msg}" }
-          end
-
-          rc = status.exitstatus
+      ActiveRecord::Base.connection_pool.release_connection
+      rc = nil
+      final_stderr = []
+      msg = nil
+      Bundler.with_clean_env do
+        rc, msg, final_stderr = run_method(Gem.ruby) do |stdin|
```

Comment:
@mkanoor You no longer need to initialize the variables you are returning here.  The return values from `run_method` will be passed back through the `Bundler.with_clean_env` call.

All you need is this:

```ruby
def self.run_ruby_method(body, preamble = nil)
  ActiveRecord::Base.connection_pool.release_connection
  Bundler.with_clean_env do
    run_method(Gem.ruby) do |stdin|
      stdin.puts(preamble.to_s)
      stdin.puts(body)
      stdin.puts(RUBY_METHOD_POSTSCRIPT) unless preamble.blank?
    end
  end
end
```